### DISCUSSION TITLE
Enhancements to the base design

### DISCUSSION THREAD

**Discussion by mebs** [mebs/CM4-NAS#4] [Ideas]
I created this project during a transition period where I had a lot of time in my hands. As much fun as it was learning how to use the software and tweaking the design to infinity, I had to cut it short and prioritize functionality before I was busy with work again.
I'll start this discussion with things I wish I had more time to add or change before ordering the first revision of the PCB, but feel free to add your own ideas!

**Comment by mebs** [comment]
Here are, in no particular order, some things that could be better with the current design:

- This is my first board, so routing is certainly not optimal, it would be great to have someone experienced look at it.
- The three mounting holes are plated with copper and covered with silkscreen all the way to the edge. Should change them to resemble the four holes under the CM4.
- ~The USB selection is a bit janky. It's a naïve implementation of a manual MUX but it's really not necessary. The same type A port could be used with a three pin header and a jumper to select if the CM4 provides power to the device or if power coming from the device should be routed to the CM4's OTG select pin.~
- The MagJack connector should be vertical so that the plugged in cable would come out perpendicular (above) to the board. An even better idea to make it easier to integrate the board in a case would be to not have the MagJack on the board at all. Rather have some sort of 16 pin connector instead. You would then use this connector to plug a cable that would lead to the MagJack that would be installed at the periphery of the case. This would also save a lot of room on the board, allowing to shrink it and add a fourth mounting hole.
- ~Only one GPIO is exposed, for the temperature sensor. This was a mistake and I should have at least exposed a few pins with headers wherever I had room. It would allow to add a button to cycle through what the display shows for example and more flexibility if you want to add automation in the future~.
- ~I should add text on the silkscreen to show what the headers pins do~.
- A great idea mentioned bellow is to add an external battery pack to the NAS. It wouldn't need to last for long, just enough for the CM4 to cleanly wrap things up when it gets a signal that power is out. Lithium cell management circuitry is widely available.
- Add more power management to the board, could start with as simple as 12V and 5V regulators.

I'll add more as I think of them.

**Comment by metropt** [comment]
Is it in your plans to place a SATA controller or power management on the board or will you keep it out of the board?

**Reply by mebs** [reply]
This is an idea I planted in the other discussion about alternative designs. I plan on keeping this one as simple as possible, but once I'm done with it and I have something that works, I think it would be really cool to have at least the power management on the board. It shouldn't be too complicated assuming a 12V input. The next step would be to add an onboard SATA controller and a PCIe switch to still allow to slot in an external card (for Wi-Fi 6 for instance).
One thing's for sure, we won't get anything until those CM4s start shipping...

**Comment by buffalojr** [comment]
Great job so far, 

Check these guys out for some more ideas: https://kobol.io/

-Their solution with the small battery back up is interesting and could benefit this design.
-Break out some of the oi for status lights (or did I miss that on your design)?
-Any thoughts on going with a aluminum case design?

Watching this as I like a pi based NAS better than the Kobol64 at the moment.

**Reply by mebs** [reply]
This Kobol NAS is drop dead gorgeous... I wish I could go with an aluminium case design but unless I go with some existing solutions (like the Orico ones), I don't really have the means to make something custom.

The battery pack is a great idea, I'll add this to the list above. It could be entirely abstracted from the board and almost work with the current design, I would just lack an input on the board to tell the CM4 that power is out and that it should wrap things up. But of course it should ideally be managed by the board itself, there is room for more components.

Speaking of which, you're right, I should've added way more GPIO headers (at least one would have allowed to use a button with the OLED screen). I really rushed the project at the end because I knew I would not have much time to put on it passed some point, so I knew I was making this mistake, but that's just a first prototype.

**Reply by mebs** [reply]
Added a discussion around case design here: https://github.com/mebs/CM4-NAS/discussions/6
